minetest/minetest
# [no squash] Migrate the Android port to SDL2

This PR migrates the Android build to SDL2. Some of the benefits are:

- Less maintenance
- Mouse support
- Controller support (still needs an implementation on the Minetest side)

Please note that since CIrrDeviceSDL doesn't support the ogles1 driver, this PR removes ogles1 support on Android.*

*It should be possible to make CIrrDeviceSDL support ogles1 by using code from https://github.com/MoNTE48/Irrlicht.

## To do

This PR is a Ready for Review.

Note that the first two commits just add/delete some boilerplate. The third commit is where it gets interesting.

## How to test

Play Minetest on Android.

If you want to see something new and cool, connect a bluetooth mouse and keyboard, disable `enable_touch` in the settings menu and play like a desktop player.
